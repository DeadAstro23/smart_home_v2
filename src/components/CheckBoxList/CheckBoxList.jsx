import React from 'react';

//forked module from GitHub
const CheckBoxList = React.createClass({
    displayName: 'CheckBoxList',

    propTypes: {
        defaultData: React.PropTypes.array,
        onChange: React.PropTypes.func
    },

    getInitialState: function() {
        return {
            data: this.props.defaultData || []
        };
    },

    handleItemChange: function(e) {
        var selectedValues = [],
            newData = [];

        this.state.data.forEach(function(item) {
            if(item.value === e.target.value) {
                item.checked = e.target.checked;
            }
            if(item.checked) {
                selectedValues.push(item.value);
            }
            newData.push(item);
        });

        this.setState({data: newData});

        if(this.props.onChange) {
            this.props.onChange(selectedValues);
        }
    },

    // uncheck all items in the list
    reset: function() {
        var newData = [];
        this.state.data.forEach(function(item) {
            item.checked = false;
            newData.push(item);
        });

        this.setState({data: newData});
    },

    render: function() {
        var options;

        options = this.state.data.map(function(item, index) {
            return (
                React.createElement("div", {key: 'chk-' + index, className: "checkbox"},
                    React.createElement("label", null,
                        React.createElement("input", {
                            type: "checkbox",
                            value: item.value,
                            onChange: this.handleItemChange,
                            onClick: this.props.onClick,
                            checked: item.checked ? true : false}), " ", item.label
                    )
                )
            );
        }.bind(this));

        return (
            React.createElement("div", null,
                options
            )
        );
    }
});

export class CheckboxListDecorated extends React.Component {
    render() {
        return (
            <div className="checkboxList">
				<span className="checkboxList__title">
					{this.props.title}
				</span>
                <CheckBoxList
                    defaultData={this.props.type}
                    onClick={this.props.onClick}
                />
            </div>
        )
    }
}