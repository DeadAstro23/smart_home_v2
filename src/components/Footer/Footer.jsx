import React from 'react';
import './Footer.less';

const Footer = () => (
    <div className="footer">
        <img
            src="https://dm8eklel4s62k.cloudfront.net/images/sky-logo-2016.svg"
            width="72"
            alt="sky_logo"
        />
        <span className="footer__copyright">Â© 2017 Sky UK</span>
    </div>
);

export default Footer;