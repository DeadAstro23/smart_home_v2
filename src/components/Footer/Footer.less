.footer {
  position: absolute;
  bottom: 0;
  padding: 20px;

  &__copyright {
    margin: 0 10px 0;
    top: -10px;
    position: relative;
  }
}