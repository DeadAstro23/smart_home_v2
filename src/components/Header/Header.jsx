import React, {Component} from 'react';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import FlatButton from 'material-ui/FlatButton';
import Drawer from 'material-ui/Drawer';


import './Header.less';

class Login extends Component {
    static muiName = 'FlatButton';

    render() {
        return (
            <FlatButton {...this.props} label="Login" />
        );
    }
}

const Logged = (props) => (
    <IconMenu
        {...props}
        iconButtonElement={
            <IconButton>
                <MoreVertIcon />
            </IconButton>
        }
        targetOrigin={{horizontal: 'right', vertical: 'top'}}
        anchorOrigin={{horizontal: 'right', vertical: 'top'}}
    >
        <MenuItem primaryText="Refresh" />
        <MenuItem primaryText="Help" />
        <MenuItem primaryText="Sign out" />
    </IconMenu>
);

Logged.muiName = 'IconMenu';

class Header extends Component {
    state = {
        logged: false,
        open: false
    };

    handleToggle = () => this.setState({
        open: !this.state.open
    });


    handleChange = (event, logged) => {
        this.setState({logged: logged});
    };

    render() {

        return (
            <div>
                <Drawer open={this.state.open}
                        docked={false}
                        width={200}
                        onRequestChange={this.handleToggle}
                >
                    <MenuItem>Menu Item</MenuItem>
                    <MenuItem>Menu Item 2</MenuItem>
                    <MenuItem>Menu Item 3</MenuItem>
                    <MenuItem>Menu Item 4</MenuItem>
                    <MenuItem>Menu Item 5</MenuItem>
                    <MenuItem>Menu Item 6</MenuItem>
                    <MenuItem>Menu Item 7</MenuItem>
                </Drawer>

                <AppBar
                    title="Microsity"
                    onLeftIconButtonTouchTap={this.handleToggle}
                    iconElementRight={this.state.logged ? <Logged /> : <Login />}
                />
            </div>
        );
    }
}

export default Header;