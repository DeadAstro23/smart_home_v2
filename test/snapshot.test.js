import React from 'react';
import Header from '../src/components/Header/Header';
import Footer from '../src/components/Footer/Footer';
import CheckBoxList from '../src/components/CheckBoxList/CheckBoxList';
import renderer from 'react-test-renderer';

it('Header renders correctly', () => {
    const app = renderer.create(
        <Header />
    ).toJSON();
    expect(app).toMatchSnapshot();
});

it('Footer renders correctly', () => {
    const app = renderer.create(
        <Footer />
    ).toJSON();
    expect(app).toMatchSnapshot();
});

it('CheckBoxList renders correctly', () => {
    const app = renderer.create(
        <CheckBoxList />
    ).toJSON();
    expect(app).toMatchSnapshot();
});
