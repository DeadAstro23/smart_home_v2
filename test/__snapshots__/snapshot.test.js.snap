exports[`test CheckBoxList renders correctly 1`] = `<div />`;

exports[`test Footer renders correctly 1`] = `
<div
  className="footer">
  <img
    alt="sky_logo"
    src="https://dm8eklel4s62k.cloudfront.net/images/sky-logo-2016.svg"
    width="72" />
  <span
    className="footer__copyright">
    Â© 2017 Sky UK
  </span>
</div>
`;

exports[`test Header renders correctly 1`] = `
<div
  className="header">
  <img
    alt="sky_logo"
    className="header__logo"
    src="https://dm8eklel4s62k.cloudfront.net/images/sky-logo-2016.svg"
    width="100" />
</div>
`;
